notte/common/parser.py

```diff
@@ -62,69 +79,128 @@ def search_pattern(text: str, tag: str) -> str | None:
         return match.group(1).strip() if match else None
 
     @override
-    def which(self, text: str) -> Literal["observe", "step", "rules"]:
-        url = self.search_pattern(text, "url")
-        action = self.search_pattern(text, "action")
-        match (bool(url), bool(action)):
-            case (True, False):
+    def which(self, text: str) -> Literal["observe", "step", "scrape", "rules"]:
+        url = self.search_pattern(text, BaseNotteParser.observe_tag)
+        action = self.search_pattern(text, BaseNotteParser.step_tag)
+        scrape = self.search_pattern(text, BaseNotteParser.scrape_data_tag)
+        match (bool(url), bool(action), bool(scrape)):
+            case (True, False, False):
                 return "observe"
-            case (False, True):
+            case (False, True, False):
                 return "step"
+            case (False, False, True):
+                return "scrape"
             case _:
                 return "rules"
 
+    @override
+    def is_done(self, text: str) -> bool:
+        return (
+            f"<{BaseNotteParser.done_tag}/>" in text or self.search_pattern(text, BaseNotteParser.done_tag) is not None
+        )
+
+    @override
+    def get_done_answer(self, text: str) -> str | None:
```

Comment:
Why do you have a `is_done` and `get_done_answer` function that are essentially the same?